matrix_sdk_base/src/models/room.rs

```diff
@@ -341,126 +333,148 @@ impl Room {
         self.encrypted.as_ref()
     }
 
-    /// Get the disambiguated display name for a member of this room.
+    /// Process the join or invite event for a new room member.
     ///
-    /// If a member has no display name set, returns the MXID as a fallback. Additionally, we
-    /// return the MXID even if there is no such member in the room.
+    /// This method should only be called on events which add new members, not those related to
+    /// existing ones.
     ///
-    /// When displaying a room member's display name, clients *must* use this method to obtain the
-    /// name instead of displaying the `RoomMember::display_name` directly. This is because
-    /// multiple members can share the same display name in which case the display name has to be
-    /// disambiguated.
-    pub fn member_display_name<'a>(&'a self, id: &'a UserId) -> Cow<'a, str> {
-        let disambiguated_name = self
-            .disambiguated_display_names
-            .get(id)
-            .map(|s| s.as_str().into());
-
-        if let Some(name) = disambiguated_name {
-            // The display name of the member is non-unique so we return a disambiguated version.
-            name
-        } else if let Some(member) = self
-            .joined_members
-            .get(id)
-            .or_else(|| self.invited_members.get(id))
-        {
-            // The display name of the member is unique so we can return it directly if it is set.
-            // If not, we return his MXID.
-            member.name().into()
-        } else {
-            // There is no member with the requested MXID in the room. We still return the MXID.
-            id.as_str().into()
-        }
-    }
-
-    fn add_member(&mut self, event: &StateEventStub<MemberEventContent>) -> bool {
+    /// Returns a tuple of:
+    ///
+    /// 1. True if the event made changes to the room's state, false otherwise.
+    /// 2. Returns a map of display name disambiguations which tells us which members need to have
+    ///    their display names disambiguated and to what.
```

Comment:
We get a map from `UserId` to bool, this doesn't tell us to what the display name needs to be disambiguated, only if there is ambiguity or not, no?

Reply:
Fixed.